### PR TITLE
instantiateRenderEntity sets up render assets on render components

### PR DESCRIPTION
**Pull request by mvaligursky**
- When an Entity hierarchy was created using GlbContainer.instantiateRenderEntity, the render component didn't have .asset property set up. This is now fixed, and those properties point to a render assets.
- Additionally, a Render class is now public, as it is a type of the resource accessible using render component's asset property.

### CODE REVIEW

File: src/resources/parser/glb-container-resource.js [hunk 1]
```diff
@@ -109,6 +109,7 @@ class GlbContainerResource {
 
             // find all components needed for this node
             let attachedMi = null;
+            let renderAssset = null;
```
Comment (slimbuck) [inline src/resources/parser/glb-container-resource.js:112]: renderAsset

Comment (mvaligursky) [reply src/resources/parser/glb-container-resource.js:112]: hah :)

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
Wait, what's the type of the resource of a render asset? e.g. `entity.render.asset`?

**Comment by mvaligursky** [comment]
> Wait, what's the type of the resource of a render asset? e.g. `entity.render.asset`?

the code I added assigns Asset type to it created here:
https://github.com/playcanvas/engine/blob/d5bef0eed6b3d1f1c8970b6d33e089139d39773e/src/resources/parser/glb-container-resource.js#L58-L65

But internally this is stored as an id on the component, and render.asset returns asset.id

**Comment by willeastcott** [comment]
I meant, is the asset's resource a `Render`? Just wondering, because that type is not public API.

**Comment by mvaligursky** [comment]
> I meant, is the asset's resource a `Render`? Just wondering, because that type is not public API.

Fair point. Making the Render class public.

**Comment by willeastcott** [comment]
Are developers expected to create `Render` objects themselves? Or do anything with them? Is the `Render` API in 'final' form?

**Comment by mvaligursky** [comment]
I'm hiding the constructor as well .. as there's not a big use case for its public use.
Otherwise only meshes properties is exposed .. and I think that's useful to have.

**Comment by slimbuck** [comment]
Do we need to make `Render` class public API?

Looks like instances of `Render` are created by the glb-parser at load time and then used by the render component internally (via `asset.resource`).

Even though the `Render` class seems simple an innocuous enough,  it's not actually possible to set a `Render` instance directly on the `RenderComponent` anyway. So seems like we can leave `Render private?

**Comment by willeastcott** [comment]
I'd probably agree with that assessment. Making `Render` public is not necessary to solve the original motivation for this PR.

**Review by slimbuck — APPROVED** [review]

**Pull request merged**